src/Evolution/Systems/ScalarTensor/BoundaryConditions/Factory.hpp

```diff
@@ -0,0 +1,64 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "Domain/BoundaryConditions/Periodic.hpp"
+#include "Evolution/Systems/CurvedScalarWave/BoundaryConditions/AnalyticConstant.hpp"
+#include "Evolution/Systems/CurvedScalarWave/BoundaryConditions/DemandOutgoingCharSpeeds.hpp"
+#include "Evolution/Systems/CurvedScalarWave/BoundaryConditions/Factory.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/BoundaryCondition.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/DemandOutgoingCharSpeeds.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/DirichletAnalytic.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/DirichletMinkowski.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/Factory.hpp"
+#include "Evolution/Systems/ScalarTensor/BoundaryConditions/BoundaryCondition.hpp"
+#include "Evolution/Systems/ScalarTensor/BoundaryConditions/ProductOfConditions.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace ScalarTensor::BoundaryConditions {
+
+static constexpr size_t Dim = 3;
+
+namespace detail {
+
+// Remove boundary condition products of DemandOutgoingCharSpeeds with
+// other types of boundary conditions
+template <typename DerivedGhCondition, typename DerivedScalarCondition>
+using ProductOfConditionsIfConsistent = tmpl::conditional_t<
+    (DerivedGhCondition::bc_type ==
+     evolution::BoundaryConditions::Type::DemandOutgoingCharSpeeds) xor
+        (DerivedScalarCondition::bc_type ==
+         evolution::BoundaryConditions::Type::DemandOutgoingCharSpeeds),
+    tmpl::list<>,
+    ProductOfConditions<DerivedGhCondition, DerivedScalarCondition>>;
+
+template <typename GhList, typename ScalarList>
+struct AllProductConditions;
+
+template <typename GhList, typename... ScalarConditions>
+struct AllProductConditions<GhList, tmpl::list<ScalarConditions...>> {
+  using type = tmpl::flatten<tmpl::list<tmpl::transform<
+      GhList, tmpl::bind<ProductOfConditionsIfConsistent, tmpl::_1,
+                         tmpl::pin<ScalarConditions>>>...>>;
+};
+
+}  // namespace detail
+
+/// Typelist of standard BoundaryConditions. For now, we only support a subset
+/// of the available boundary conditions
+using subset_standard_boundary_conditions_gh =
+    tmpl::list<gh::BoundaryConditions::DemandOutgoingCharSpeeds<Dim>,
+               gh::BoundaryConditions::DirichletAnalytic<Dim>,
+               gh::BoundaryConditions::DirichletMinkowski<Dim>>;
```

Comment:
We don't need the `DirichletMinkowski` anymore. That was resolved by being able to set the analytic solution/data to set in the BC. It should be deleted from GH, so I would just omit it here to make the work of removing it easier.